core/src/main/java/com/linecorp/armeria/client/endpoint/dns/DnsEndpointGroupBuilder.java

```diff
@@ -153,6 +159,35 @@ final DefaultDnsResolver buildResolver(Consumer<DnsNameResolverBuilder> customiz
                                      hostsFileEntriesResolver());
     }
 
+    /**
+     * Adds the {@link DnsQueryListener}s that listens to the result of querying {@link DnsRecord}s.
+     * If no {@link DnsQueryListener} is configured, {@link DnsQueryListener#of()} is used by default
+     */
+    @UnstableApi
+    public DnsEndpointGroupBuilder addDnsQueryListeners(
+            Iterable<? extends DnsQueryListener> dnsQueryListeners) {
+        requireNonNull(dnsQueryListeners, "dnsQueryListeners");
+        for (DnsQueryListener listener: dnsQueryListeners) {
+            this.dnsQueryListeners.add(listener);
+        }
+        return this;
+    }
+
+    /**
+     * Adds the {@link DnsQueryListener} that listens to the result of querying {@link DnsRecord}s.
+     * If no {@link DnsQueryListener} is configured, {@link DnsQueryListener#of()} is used by default
+     */
+    @UnstableApi
+    public DnsEndpointGroupBuilder addDnsQueryListeners(DnsQueryListener... dnsQueryListeners) {
+        requireNonNull(dnsQueryListeners, "dnsQueryListeners");
+        this.dnsQueryListeners.addAll(Arrays.asList(dnsQueryListeners));
+        return this;
```

Comment:
Delegate to `Iterable` version?
```suggestion
        return dnsQueryListeners(ImmutableList.copyOf(dnsQueryListeners));
```